sklearn/preprocessing/imputation.py

```diff
@@ -374,3 +385,464 @@ def transform(self, X):
+            Input data, where "n_samples" is the number of samples and
+            "n_features" is the number of features.
+
+        Returns
+        -------
+        X : array-like, shape (n_samples, n_features)
+            The imputed input data.
+        """
+        if self.estimator is None:
+            from ..linear_model import BayesianRidge
+            self.estimator_ = BayesianRidge()
+        else:
+            self.estimator_ = clone(self.estimator)
+
+        # check that the estimator's predict method has return_std argument
+        if not ('return_std' in signature(self.estimator_.predict).parameters):
+            raise Exception("The regression estimator is %s and its predict "
+                            "method does not support 'return_std=True'. "
+                            "This is required for MICE." %
+                            type(self.estimator_))
+
+        # pre-processing of X
+        X = check_array(X, dtype=np.float64, force_all_finite=False)
+        X = np.asarray(X, order="F")
+
+        # parse min and max values
+        self.min_value_ = np.nan if self.min_value is None else self.min_value
+        self.max_value_ = np.nan if self.max_value is None else self.max_value
+
+        # initial imputation
+        mask_missing_values = _get_mask(X, self.missing_values)
+        self.initial_imputer_ = Imputer(missing_values=self.missing_values,
+                                        strategy=self.initial_fill_method,
+                                        axis=0)
+        X_filled = self.initial_imputer_.fit_transform(X)
+        self._val_inds = self.initial_imputer_._valid_statistics_inds
+        X = X[:, self._val_inds]
+        mask_missing_values = mask_missing_values[:, self._val_inds]
+
+        # perform imputations
+        self._trained_estimator_triplets = []
+        n_samples, n_features = X_filled.shape
+        total_rounds = self.n_burn_in + self.n_imputations
+        results_list = []
+        if self.verbose:
+            print("[MICE] Completing matrix with shape %s" % (X.shape,))
+            start_t = time()
+            mice_msg = '[MICE] Ending imputation round '
+        for m in range(total_rounds):
+            # order in which to impute
+            ordered_inds = self._get_ordered_inds(mask_missing_values)
+
+            # abs_correlation matrix is used to choose a subset of other
+            # features to impute from  f
+            abs_corr_mat = self._get_abs_correlation_matrix(X_filled)
+
+            # Fill in each feature in the order of ordered_inds
+            for this_feat_ind in ordered_inds:
+                i = this_feat_ind  # the tyranny of pep8
+                other_feat_inds = self._get_other_feat_inds(n_features,
+                                                            i,
+                                                            abs_corr_mat)
+                X_filled, estimator = self._impute_one_feature(
```

Comment:
I think that could be simplified by a list comprehension but some code will move in the meanwhile so let see